src/network/networkd-link.c

```diff
@@ -256,6 +257,38 @@ void link_update_operstate(Link *link, bool also_update_master) {
         else
                 operstate = LINK_OPERSTATE_ENSLAVED;
 
+        /* Only determine online state for managed links with RequiredForOnline=yes */
+        if (!link->network || link->network->unmanaged || !link->network->required_for_online)
```

Comment:
Sorry, this is my fault. I previously said that you also need to check `unmanaged` flag, but it is not necessary. As, you can see e.g. `link_initialized_and_synced()`, if `network->unmanaged` is set, we do not assign the `network` object to `link`. So,
```
if (!link->network || !link->network->required_for_online)
```
is enough. Sorry.